renderite/Bootstrap.zig

```diff
@@ -0,0 +1,137 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const zinterprocess = @import("zinterprocess");
+const Queue = zinterprocess.Queue;
+
+const InitSettings = @import("InitSettings.zig");
+
+const log = std.log.scoped(.Bootstrap);
+
+const Bootstrap = @This();
+
+queue_in: ?Queue,
+queue_out: ?Queue,
+child: ?std.process.Child,
+init_settings: InitSettings,
+
+pub fn init(args: []const []const u8, gpa: std.mem.Allocator) !Bootstrap {
+    if (args.len > 1 and std.mem.eql(u8, args[1], "-QueueName")) {
+        // If the renderer is launching us directly, we need no special logic.
+        return .{
+            .queue_in = null,
+            .queue_out = null,
+            .child = null,
+            .init_settings = try InitSettings.init(args),
+        };
+    } else {
+        const prefix, const queue_in, const queue_out = try initBootstrapQueues();
+        const child = try startResonite(prefix, gpa);
+
+        // wait for resonite's bootstrapper hello message
+        const message = try queue_in.dequeue(gpa);
+        log.debug("Received queue message: {s}", .{message});
+
+        var iterator = std.mem.splitAny(u8, message, " ");
+        const max_part = 4;
+        var parts: [max_part][]const u8 = undefined;
+
+        var i: usize = 0;
+        while (iterator.next()) |part| {
+            std.debug.assert(i < max_part);
+
+            parts[i] = part;
+            i += 1;
+        }
+
+        const init_settings = try InitSettings.init(parts[0..max_part]);
+
+        const pid = switch (builtin.target.os.tag) {
+            .windows => std.os.windows.GetCurrentProcessId(),
+            else => std.os.linux.getpid(),
+        };
+
+        var msg_buf: [32]u8 = undefined;
+        const msg = try std.fmt.bufPrint(&msg_buf, "RENDERITE_STARTED:{d}", .{pid});
+
+        try queue_out.enqueue(msg);
+
+        const bootstrap: Bootstrap = .{
+            .queue_in = queue_in,
+            .queue_out = queue_out,
+            .child = child,
+            .init_settings = init_settings,
+        };
+
+        return bootstrap;
+    }
+}
+
+fn initBootstrapQueues() !std.meta.Tuple(&.{ []const u8, Queue, Queue }) {
```

Comment:
```suggestion
fn initBootstrapQueues() !struct { []const u8, Queue, Queue } {
```

`std.meta.Tuple` is mostly when the list is generated at comptime, by omitting the names of struct fields, you create a tuple (this should also make ZLS much happier).

Reply:
Hmm, I see. Should I do this in the generator also for packed bools?